Sources/Lang.php

```diff
@@ -662,29 +701,51 @@ public static function sentenceList(array $list): string
 	public static function numberFormat(int|float|string $number, ?int $decimals = null): string
 	{
 		if (is_string($number)) {
-			$number = intval($number);
+			$number = $number + 0;
```

Comment:
Any reason for that? I know it's the same result, but it's just loose type converting.  I'm used to (from my job working in other languages) that you can't do that to force them to convert over like that without a casting.  Don't mean to get up on it, just that I'm used to casting things and its not allowed to mix incompatible types.

Reply:
Because $number might be '1.25'

Reply:
Ahh, floatval might be the best option then.

Reply:
Not if $number is something like '1'. 

For the purposes of MessageFormat strings, 1 and 1.0 are distinct values that need different handling in many languages. Therefore, neither intval() nor floatval() are suitable. Adding zero is the simplest, cleanest way to get the necessary behaviour in PHP.

Reply:
Might be a good place for is_numeric instead of just a is_string check.

Reply:
Good idea.